jgrapht-ext/src/main/java/org/jgrapht/ext/DOTExporter.java

```diff
@@ -266,6 +285,20 @@ private String getVertexID(V v)
                 + "' is not valid with respect to the .dot language");
     }
 
+    /**
+     * Test if the ID candidate is a valid ID.
+     *
+     * @param idCandidate the ID candidate.
+     *
+     * @return <code>true</code> if it is valid; <code>false</code> otherwise.
+     */
+    public static boolean isValidID(final String idCandidate) {
+        return ALPHA_DIG.matcher(idCandidate).matches()
+                || DOUBLE_QUOTE.matcher(idCandidate).matches()
+                || DOT_NUMBER.matcher(idCandidate).matches()
+                || HTML.matcher(idCandidate).matches();
+    }
+    
```

Comment:
I made it package-protected in `DOTUtils`, but I think that this could be useful for people who wants to check if an ID is valid outside the import/export tasks, without compile the patterns themselves.